### PR TITLE
Update build dependencies

### PR DESCRIPTION
**Pull request by mfcrocker**
# What and why?
Our build dependencies haven't been updated for a while.

# How to test?
Upload a sample

# Jira
RAS-902

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
         run: |
           export LD_LIBRARY_PATH=/usr/local/lib
           mvn fmt:check
-          mvn clean verify
+          mvn clean -DskipITs verify
```
Comment (pricem14pc) [inline .github/workflows/main.yml:57]: The existing `CachingTestIT` tests have never run in this repo. Prior to this update build dependencies bump you will see the failsafe plugin NOT running anything
> Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
The bump of these build dependencies in this PR has had the effect of now running the tests. However, these tests looks like they're always been broken and fail.
I have made a number of changes in this branch to address a number of the errors they generate, but they need revisiting completely in a new card. It is not obvious what the original intention was for these integration tests. They don't appear to be correct.

As such, I am temporarily suppressing them in the PR build. That is no worse than then not previously running.

File: docker-compose.yml [hunk 2]
```diff
@@ -1,38 +1,67 @@
 
 version: '2'
 services:
+
   postgres:
     container_name: postgres-it
-    image: sdcplatform/ras-rm-docker-postgres
+    image: eu.gcr.io/ons-rasrmbs-management/postgres
+    command: postgres -c 'max_connections=1000'
```
Comment (pricem14pc) [inline docker-compose.yml:8]: Inelegant, but this fixes the `Too many clients already` error. That error appears to be due to failsafe generating multiple connections per test and not purging or reusing the existing connections between tests.

File: pom.xml [hunk 3]
```diff
@@ -17,7 +17,8 @@
 		<spring.cloud.gcp.version>1.2.8.RELEASE</spring.cloud.gcp.version>
 		<common.version>10.49.4</common.version>
 		<java.version>17</java.version>
-		<surefire.version>2.20</surefire.version>
+		<surefire.version>3.2.2</surefire.version>
+		<skipSurefire>false</skipSurefire>
```
Comment (pricem14pc) [inline pom.xml:21]: Adding this to make it quicker to run maven builds when investigating the failsafe integration tests during development

File: pom.xml [hunk 4]
```diff
@@ -443,6 +445,7 @@
 					<version>${surefire.version}</version>
 					<configuration>
 						<threadCount>1</threadCount>
+						<skip>${skipSurefire}</skip>
```
Comment (pricem14pc) [inline pom.xml:448]: See comment above

File: src/main/resources/application-test.yml [hunk 5]
```diff
@@ -24,4 +25,8 @@ collection-exercise-svc:
 
 gcp:
   project: "test"
+  sampleSummaryActivationSubscription: "test_activation_subscription"
+  sampleSummaryActivationStatusTopic: "test_activation_topic"
   caseNotificationTopic: "test_topic"
+  collectionExerciseEndSubscription: "test_collection_exercise_end_sample"
+  sampleSummaryDeadLetterSubscription: "test_dead_letter_subscription"
```
Comment (pricem14pc) [inline src/main/resources/application-test.yml:32]: ApplicationContext was failing due to missing pubsub config

File: src/test/resources/uk/gov/ons/ctp/response/sample/config/CachingTestIT.CollectionInstrumentDTO.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+      "8f078c99-2843-47c6-9c57-13e5966fbc9e"
+    ],
+    "RU_REF": [
+      "test_ru_ref"
+    ]
+  },
+  "file_name": "file_name",
+  "id": "7574283a-d1fd-49df-b684-d7b201e5748a",
+  "surveyId": "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87"
+}]
```
Comment (pricem14pc) [inline src/test/resources/uk/gov/ons/ctp/response/sample/config/CachingTestIT.CollectionInstrumentDTO.json:13]: ApplicationContext was failing to load due to this missing resource

File: src/test/resources/uk/gov/ons/ctp/response/sample/config/CachingTestIT.CollectionInstrumentDTO2.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+      "8f078c99-2843-47c6-9c57-13e5966fbc9e"
+    ],
+    "RU_REF": [
+      "test_ru_ref"
+    ]
+  },
+  "file_name": "file_name",
+  "id": "7574283a-d1fd-49df-b684-d7b201e5748a",
+  "surveyId": "cb0711c3-0ac8-41d3-ae0e-567e5ea1ef88"
+}]
```
Comment (pricem14pc) [inline src/test/resources/uk/gov/ons/ctp/response/sample/config/CachingTestIT.CollectionInstrumentDTO2.json:13]: ApplicationContext was failing to load due to this missing resource

**Review by arroyoAle — APPROVED** [review]
changes look good, integration tests don't run in GHA and sample svc works  when deployed

**Review by LJBabbage — APPROVED** [review]
Tested and worked fine

**Pull request merged**